### PR TITLE
migrate NAT gateway to cloudformation

### PR DESCRIPTION
**Pull request by fgimenez**
Towards https://github.com/giantswarm/giantswarm/issues/2023

### CODE REVIEW

File: service/resource/legacyv2/adapter/adapter.go [hunk 1]
```diff
@@ -37,9 +37,10 @@ type Adapter struct {
 
 	launchConfigAdapter
 	autoScalingGroupAdapter
-	outputsAdapter
+	natGatewayAdapter
 	workerPolicyAdapter
 	recordSetsAdapter
+	outputsAdapter
```
Comment (rossf7) [inline service/resource/legacyv2/adapter/adapter.go]: This ordering with outputs last is much nicer. <3

File: service/templates/cloudformation/nat_gateway.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{{define "nat_gateway"}}
+  NATGateway:
+    Type: AWS::EC2::NatGateway
+    Properties:
+      AllocationId:
+        Fn::GetAtt:
+        - NATEIP
+        - AllocationId
+      SubnetId: {{.PublicSubnetID}}
```
Comment (rossf7) [inline service/templates/cloudformation/nat_gateway.yaml]: Can we add a Name tag here with the cluster ID? 

We do that in the old AWS resources. We get Cloud Formation tags added automatically with the stack name that has the cluster ID. But the name tag is useful in the AWS console.

Comment (fgimenez) [reply service/templates/cloudformation/nat_gateway.yaml]: Sure, done. I've taken the opportunity to promote `ClusterID` to the top level adapter (it was used in ASG too). Also now we don't try to delete NAT gw if on CloudFormation.

Comment (rossf7) [reply service/templates/cloudformation/nat_gateway.yaml]: Thanks, and yes I like that shared fields are moved to the top level adapter.

**Pull request merged**